sdk/python/feast/feature_table.py

```diff
@@ -153,7 +166,7 @@ def stream_source(self):
         return self._stream_source
 
     @stream_source.setter
-    def stream_source(self, stream_source: DataSource):
+    def stream_source(self, stream_source: DataSourceProto):
```

Comment:
Why does this take a proto?

Reply:
Similar to expecting FeatureSpecV2, we're expecting DataSourceProto when generating FeatureTableSpec in `to_proto` method. This allows the user to supply just the native DataSource class when using the SDK.

Reply:
I've cleaned this up, to shift conversion inside `to_proto` method, this makes typings and initialization of class object code alot cleaner.